src/swap/letsexchange.js

```diff
@@ -0,0 +1,262 @@
+// @flow
+
+import { lt } from 'biggystring'
+import { asObject, asString } from 'cleaners'
+import {
+  type EdgeCorePluginOptions,
+  type EdgeCurrencyWallet,
+  type EdgeSpendInfo,
+  type EdgeSwapInfo,
+  type EdgeSwapPlugin,
+  type EdgeSwapQuote,
+  type EdgeSwapRequest,
+  type EdgeTransaction,
+  SwapBelowLimitError,
+  SwapCurrencyError
+} from 'edge-core-js/types'
+
+import {
+  type InvalidCurrencyCodes,
+  checkInvalidCodes,
+  getCodesWithMainnetTranscription,
+  makeSwapPluginQuote
+} from '../swap-helpers.js'
+import { asOptionalBlank } from './changenow'
+
+const pluginId = 'letsexchange'
+const swapInfo: EdgeSwapInfo = {
+  pluginId,
+  displayName: 'LetsExchange',
+  supportEmail: 'support@letsexchange.io'
+}
+
+const orderUri = 'https://letsexchange.io/?exchangeId='
+const uri = 'https://api.letsexchange.io/api/v1/'
+
+const expirationMs = 1000 * 60 * 20
+
+const asQuoteInfo = asObject({
+  transaction_id: asString,
+  status: asString,
+  coin_from: asString,
+  coin_to: asString,
+  coin_from_network: asString,
+  coin_to_network: asString,
+  deposit_amount: asString,
+  withdrawal_amount: asString,
+  deposit: asString,
+  deposit_extra_id: asOptionalBlank(asString),
+  withdrawal: asString,
+  withdrawal_extra_id: asOptionalBlank(asString),
+  rate: asString,
+  fee: asString,
+  return: asString,
+  hash_in: asOptionalBlank(asString),
+  hash_out: asOptionalBlank(asString)
+})
+
+const dontUseLegacy = {
+  DGB: true
+}
+
+const INVALID_CURRENCY_CODES: InvalidCurrencyCodes = {
+  from: {},
+  to: {
+    zcash: ['ZEC']
+  }
+}
+
+// Network names that don't match parent network currency code
+const MAINNET_CODE_TRANSCRIPTION = {
+  ethereum: 'ERC20',
```

Comment:
I tested the plugin and I wasn't able to get a quote for BTC -> ETH and it failed on the /info call. It appears the issue is the network_to and network_from fields have multiple possible in the LetsExchange api for each network. For example, when trading for ETH the network_to is supposed to be 'ETH' but when trading an ERC20 token like USDT the network_to is supposed to be 'ERC20'. There will need to be logic to determine if the main chain currency code or it's token type should be used in the network fields.

Reply:
Fixed it. But I didn't find ETH in the app.
![image width=200px](https://user-images.githubusercontent.com/80786374/163682986-3d91b198-626c-4408-871a-ee1ad25e26bf.png)

Reply:
There was a recent bug affecting search results I suspect you're running a version of edge-react-gui that still has it (issue was fixed ~10 days ago). 

The change you made doesn't address the issue I raised. I was describing how the LetsExchange API works and the logic necessary to accurately request quotes. For example:

1) User wants to trade ETH so the relevant network field needs to be ETH.
2) User wants to trade USDT (on ethereum) so the relevant network field needs to be ERC20

The logic needs to be, if the currency being traded is the main chain currency, then use that currency code as the network (you can use request.fromWallet.currencyInfo.currencyCode and request.toWallet.currencyInfo.currencyCode to do the comparison). Else, it's a token and you'll need to use the MAINNET_CODE_TRANSCRIPTION map to get the appropriate network name. 

Revert the change you make and apply the logic above and you should be good to go.

Reply:
Hi! I did a "git pull" for edge-react-gui. My wallet list got bigger but ETH is still missing.

Reply:
How can I select USDT (on Ethereum) or any another coin on Ethereum in the app?
I only see the name of the coin, token. I don't see the network name

Reply:
This is what i got when i chose USDT(on AVAX)
<img width="200" alt="image" src="https://user-images.githubusercontent.com/80786374/163970033-e387f29a-f8d5-4495-b2f6-4334b4235227.png">

Reply:
This is what i got when try to chose USDT(on AVAX)

<img width="200" alt="image" src="https://user-images.githubusercontent.com/80786374/163971851-51ebc2df-d136-4d7a-9d49-caedf0e2cf04.png">

Reply:
I did new PR with all fixes.
https://github.com/EdgeApp/edge-exchange-plugins/pull/190